lib/stimulus_reflex/reflex.rb

```diff
@@ -76,6 +77,7 @@ def request
       )
       path_params = Rails.application.routes.recognize_path_with_request(req, url, req.env[:extras] || {})
       req.env.merge(ActionDispatch::Http::Parameters::PARAMETERS_KEY => path_params)
+      req.env["action_dispatch.request.parameters"] = @params
```

Comment:
What happens if we modify to this?

```ruby
req.env.merge(ActionDispatch::Http::Parameters::PARAMETERS_KEY => path_params.merge(@params))
```

Reply:
This was the first thing I tried yesterday. Because of how [Rails works under the hood](https://github.com/rails/rails/blob/777595212e5bda9f804da7cbb46d5775918f3457/actionpack/lib/action_dispatch/http/parameters.rb#L66), this will only assign the `path_parameters` and ignore the `request_parameters`.

This leaves us with just path parameters when all is said and done, hence the extra env assignment.

Reply:
Makes sense.